datumaro/plugins/velodynepoints_format/converter.py

```diff
@@ -0,0 +1,345 @@
+            self._context._builtin_attrs
+
+    def _write_item(self, item, index):
+        if not self._context._reindex:
+            index = cast(item.attributes.get('frame'), int, index)
+        image_info = OrderedDict([("id", str(index)), ])
+        if isinstance(item.pcd, str) and osp.isfile(item.pcd):
+            path = item.pcd.replace(os.sep, '/')
+            filename = path.rsplit("/", maxsplit=1)[-1]
+        else:
+            filename = self._context._make_pcd_filename(item)
+        image_info["name"] = filename
+
+        if item.has_pcd:
+            if self._context._save_images:
+                velodyne_dir = osp.join(self._context._save_dir, "velodyne_points")
+                self._context._image_dir = osp.join(velodyne_dir, "data")
+
+                self._context._save_pcd(item,
+                                        osp.join(self._context._image_dir, filename))
+
+                if item.related_images:
+                    related_dir = osp.join(self._context._save_dir, f"image_0{index}")
+
+                    for i, related_image in enumerate(item.related_images):
+                        path = osp.join(related_dir, "data")
+                        os.makedirs(path, exist_ok=True)
+
+                        try:
+                            name = related_image["name"]
+                            image = related_image["image"]
+                        except AttributeError:
+                            name = f"{i}.jpg"
+
+                        path = osp.join(path, name)
+
+                        if isinstance(image, ByteImage):
+                            with open(path, 'wb') as f:
+                                f.write(image.get_bytes())
+                        else:
+                            save_image(path, image.data)
+
+        else:
+            log.debug("Item '%s' has no image info", item.id)
+
+
+class VelodynePointsConverter(Converter):
+    DEFAULT_IMAGE_EXT = ".pcd"
+
+    @classmethod
+    def build_cmdline_parser(cls, **kwargs):
+        parser = super().build_cmdline_parser(**kwargs)
+        parser.add_argument('--reindex', action='store_true',
+            help="Assign new indices to frames (default: %(default)s)")
+        parser.add_argument('--allow-undeclared-attrs', action='store_true',
+            help="Write annotation attributes even if they are not present in "
+                "the input dataset metainfo (default: %(default)s)")
+        return parser
+
+    def __init__(self, extractor, save_dir, reindex=False,
+            allow_undeclared_attrs=False, **kwargs):
+        super().__init__(extractor, save_dir, **kwargs)
+
+        self._reindex = reindex
+        self._builtin_attrs = VelodynePointsPath.BUILTIN_ATTRS
+        self._allow_undeclared_attrs = allow_undeclared_attrs
+
+    def apply(self):
+
+        for subset_name, subset in self._extractor.subsets().items():
+
+            with open(osp.join(self._save_dir, 'tracklets.xml'), 'w') as f:
```

Comment:
Multiple subsets will be written to the same file, overriding it.

Reply:
In this case, all the subset data is written to the tracklets.xml file in accordance to the kitti format.